packages/@uppy/companion/src/companion.js

```diff
@@ -102,6 +102,11 @@ module.exports.app = (options = {}) => {
   app.use('/s3', s3(options.providerOptions.s3))
   app.use('/url', url())
 
+  // hooks directly into youtube-dl while avoiding Uppy's meta endpoints
+  // previously, we tried to implement this w/in url controller but had tough to debug problems
+  app.post('/youtube/download', controllers.youtube.download)
+  app.post('/youtube/upload', controllers.youtube.upload)
```

Comment:
DL/UL routes are separate mainly for better UX. Currently, to upload via tusd, we're using `uppy.addFile()`. This closes the plugin UI on submission and shows errors in a toast (compared to showing an error message in the plugin UI). Also, for longer videos, you actually see the filesize progress upload after the initial download is finished.

We're already needing to download the file completely before using tusd, so this isn't too much different.